godot-core/src/builtin/quaternion.rs

```diff
@@ -174,21 +186,73 @@ impl Quaternion {
         inv_factor * self + new_factor * to
     }
 
-    // pub fn spherical_cubic_interpolate(self, b: Self, pre_a: Self, post_b: Self, weight: real) -> Self {}
-    // TODO: Implement godot's function in Rust
-    /*
-        pub fn spherical_cubic_interpolate_in_time(
-            self,
-            b: Self,
-            pre_a: Self,
-            post_b: Self,
-            weight: real,
-            b_t: real,
-            pre_a_t: real,
-            post_b_t: real,
-        ) -> Self {
+    /// Performs spherical cubic interpolation.
+    ///
+    /// # Panics
+    /// If the provided Quaternions are not normalized.
+    pub fn spherical_cubic_interpolate(
+        self,
+        b: Self,
+        pre_a: Self,
+        post_b: Self,
+        weight: real,
+    ) -> Self {
+        let interpolated =
+            self.as_inner()
+                .spherical_cubic_interpolate(b, pre_a, post_b, weight.as_f64());
+        // Godot returns default if you give it Quaternions that are not normalized. This means we can check for default
+        // Then check if we should panic.
+        if interpolated == Quaternion::default()
+            && (!b.is_normalized()
+                || !pre_a.is_normalized()
+                || !post_b.is_normalized()
+                || !self.is_normalized())
+        {
+            panic!(
+                "Attempted spherical cubic interpolation on Quaternions that are not normalized."
+            );
+        }
+        interpolated
+    }
+
+    /// Performs spherical cubic interpolation in time.
+    ///
+    /// # Panics
+    /// If the provided Quaternions are not normalized.
+    #[allow(clippy::too_many_arguments)]
+    pub fn spherical_cubic_interpolate_in_time(
+        self,
+        b: Self,
+        pre_a: Self,
+        post_b: Self,
+        weight: real,
+        b_t: real,
+        pre_a_t: real,
+        post_b_t: real,
+    ) -> Self {
+        let interpolated = self.as_inner().spherical_cubic_interpolate_in_time(
+            b,
+            pre_a,
+            post_b,
+            weight.as_f64(),
+            b_t.as_f64(),
+            pre_a_t.as_f64(),
+            post_b_t.as_f64(),
+        );
+        // Godot returns default if you give it Quaternions that are not normalized. This means we can check for default
+        // Then check if we should panic.
+        if interpolated == Quaternion::default()
+            && (!b.is_normalized()
+                || !pre_a.is_normalized()
+                || !post_b.is_normalized()
+                || !self.is_normalized())
```

Comment:
As this condition appears more than once, a short private function could be worth it.